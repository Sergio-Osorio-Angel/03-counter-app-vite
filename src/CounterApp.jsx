import { useState } from 'react';
import  PropTypes from "prop-types";

export function CounterApp({value}) {

    // value sería el valor inicial del hook
    const [counter, setCounter] = useState(value);

    // Si la función utiliza algún prop, debe de ir dentro del functional component
    // De lo contrario, debería de ir fuera del functional component
    function handleAdd(event) {
        setCounter(counter + 1);
    }
    function handleSubstract(event) {
        setCounter(counter -1);
    }
    function handleReset(event) {
        setCounter(value);
    }

    return (
        <>
        <h1>CounterApp</h1>
        <h2>{counter}</h2>
        {/* <button onClick={ (event) => handleAdd(event)}>+1</button> */}
        <button onClick={ handleAdd }>+1</button>
        <button onClick={ handleSubstract }>-1</button>
        <button aria-label='btn-reset' onClick={ handleReset }>Reset</button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}