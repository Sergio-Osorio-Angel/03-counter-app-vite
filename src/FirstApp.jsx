import  PropTypes from "prop-types";

export function FirstApp({name, age}) {

    // const name = 'Sergio Osorio Angel';

    return (
    <>
        <h1  data-testid="test-saludo">{getSaludo()}, {name}</h1>
        <h2>Edad: {age}</h2>
    </>
    )
}

function getSaludo() {
    return (<>Hola Mundo</>)
}

FirstApp.propTypes = {
    name: PropTypes.string,
    age: PropTypes.number.isRequired
}

FirstApp.defaultProps = {
    name: 'Usuario Guess',
    age: 0
}