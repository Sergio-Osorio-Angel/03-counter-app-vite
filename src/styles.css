html, body {
    background-color: #21232A;
    color: white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 50px;
}
button {
    padding: 5px 20px;
    text-align: center;
    font-size: 1.3rem;
    border: none;
    margin-right: 5px;
}