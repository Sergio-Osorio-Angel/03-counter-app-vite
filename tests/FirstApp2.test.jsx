import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Test - FirstApp', () => {

    const name = 'Sergio Osorio Angel';
    const age = 21;

    test('Match con el Snapshot', () => {
        const {container} = render(<FirstApp name={name} age={age}></FirstApp>);
        expect(container).toMatchSnapshot();
    })

    test('Debe mostrar el nombre en un h1', () => {
        render(<FirstApp name={name} age={age}></FirstApp>);
        expect(screen.getByTestId('test-saludo').innerHTML.trim()).toContain(name);

        // Imprime todo el <body></body> del renderizado alamacenado
        // screen.debug();
    })

    test('Debe mostrar el nombre en un H1', () => {
        render(<FirstApp name={name} age={age}></FirstApp>);
        expect(screen.getByRole('heading', {level:1}).innerHTML).toContain(name);
    })

    test('Debe de mostrar la edad enviada por el prop', () => {
        render(<FirstApp name={name} age={age}></FirstApp>);
        expect(screen.getByRole('heading', {level:2}).innerHTML).toContain(''+age);
    })

})