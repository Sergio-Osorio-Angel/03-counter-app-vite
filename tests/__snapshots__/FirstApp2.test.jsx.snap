// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test - FirstApp Match con el Snapshot 1`] = `
<div>
  <h1
    data-testid="test-saludo"
  >
    Hola Mundo
    , 
    Sergio Osorio Angel
  </h1>
  <h2>
    Edad: 
    21
  </h2>
</div>
`;
